src/Auth0.OidcClient.Core/Tokens/IdTokenValidator.cs

```diff
@@ -0,0 +1,140 @@
+        /// <param name="signatureVerifier">Optional <see cref="ISignatureVerifier"/> to perform signature verification and token extraction. If unspecified
+        /// <see cref="AsymmetricSignatureVerifier"/> is used against the <paramref name="required"/> Issuer.</param>
+        /// <exception cref="IdTokenValidationException">Exception thrown if <paramref name="rawIDToken"/> fails to
+        /// meet the requirements specified by <paramref name="required"/>.
+        /// </exception>
+        /// <returns><see cref="Task"/> that will complete when the token is validated.</returns>
+        internal static async Task AssertTokenMeetsRequirements(this IdTokenRequirements required, string rawIDToken, DateTime pointInTime, ISignatureVerifier signatureVerifier = null)
+        {
+            if (string.IsNullOrWhiteSpace(rawIDToken))
+            {
+                if (required.TokenIsRequired)
+                    throw new IdTokenValidationException("ID token is required but missing.");
+                return;
+            }
+
+            if (signatureVerifier == null) { 
+                var jsonWebKeys = await JsonWebKeys.GetForIssuer(required.Issuer);
+                signatureVerifier = new AsymmetricSignatureVerifier(jsonWebKeys.Keys);
+            }
+
+            var token = signatureVerifier.VerifySignature(rawIDToken);
+
+            var epochNow = EpochTime.GetIntDate(pointInTime);
+
+            // Issuer
+            if (string.IsNullOrWhiteSpace(token.Issuer))
+                throw new IdTokenValidationException("Issuer (iss) claim must be a string present in the ID token.");
+            if (token.Issuer != required.Issuer)
+                throw new IdTokenValidationException($"Issuer (iss) claim mismatch in the ID token; expected \"{required.Issuer}\", found \"{token.Issuer}\".");
+
+            // Subject
+            if (string.IsNullOrWhiteSpace(token.Subject))
+                throw new IdTokenValidationException("Subject (sub) claim must be a string present in the ID token.");
+
+            // Audience
+            var audienceCount = token.Audiences.Count();
+            if (audienceCount == 0)
+                throw new IdTokenValidationException("Audience (aud) claim must be a string or array of strings present in the ID token.");
+            if (!token.Audiences.Contains(required.Audience))
+                throw new IdTokenValidationException($"Audience (aud) claim mismatch in the ID token; expected \"{required.Audience}\" but was not one of \"{String.Join(", ", token.Audiences)}\".");
+
+            // Expires at
+            var exp = GetEpoch(token.Claims, JwtRegisteredClaimNames.Exp);
+            if (exp == null)
+                throw new IdTokenValidationException("Expiration Time (exp) claim must be an integer present in the ID token.");
+            var expiration = exp + required.Leeway.TotalSeconds;
+            if (epochNow >= expiration)
```

Comment:
```suggestion
            if (epochNow > expiration)
```

... according to the sample

Reply:
I'm OK if it's for consistency. But I wouldn't be nit-picking these ones. Specially because we already set a default leeway of 60 secs that should cover most of the use cases.